presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkModule.java

```diff
@@ -485,7 +486,8 @@ protected void setup(Binder binder)
 
         binder.bind(RemoteTaskStats.class).in(Scopes.SINGLETON);
         newExporter(binder).export(RemoteTaskStats.class).withGeneratedName();
-        binder.bind(NativeExecutionProcessFactory.class).in(Scopes.SINGLETON);
+        binder.bind(PrestoSparkNativeExecutionProcessFactory.class).in(Scopes.SINGLETON);
+        newOptionalBinder(binder, NativeExecutionProcessFactory.class).setDefault().to(PrestoSparkNativeExecutionProcessFactory.class).in(Scopes.SINGLETON);
```

Comment:
Since, you are already providing an implementation (PrestoSparkNativeExecutionProcessFactory), we can always bind `NativeExecutionProcessFactory` to this implementation here. Right?

So, this will just become:
```
binder.bind( NativeExecutionProcessFactory.class).to(PrestoSparkNativeExecutionProcessFactory.class).in(Scopes.SINGLETON);
```

Reply:
Tried that, if we do it in this way here,  it said the `NativeExecutionProcessFactory.class` is already binded if we want to do the bind again for the internal implementation. that's why I use the OptionalBinder.